### DISCUSSION TITLE
Help with Ubisys J1?

### DISCUSSION THREAD

**Discussion by Sailing74** [Koenkk/zigbee2mqtt#16572] [Devices]
Hi, I'm trying to get a Ubisys J1 working here. When I tell the J1 to open, it triggers the motor for about half a second and then stops again. I tried calibration:

`mosquitto_pub -h localhost -t 'zigbee2mqtt/Fstr_Vorratskeller/set' -u xxx -P xxx -d -m '{
  "configure_j1": {
    "windowCoveringType": 0,
    "calibrate": 1,
    "open_to_closed_s": 15,
    "closed_to_open_s": 15
  }
}'`

This triggers four of these tiny half-a-second movements, but changes nothing else. If I attach buttons to the J1, I see the same behavior: long and short presses only make the motor move for half a second.

Any idea what I could try would be highly appreciated.

Thanks!

**Comment by sjorge** [comment]
You cant manually configure and calibrate together.

So either let it so the calibration (takes a while)

'{ "configure_j1": { "windowCoveringType": 0, "calibrate": 1}'

Or set open_to_closed_s and closed_to_open_s.

**Comment by Sailing74** [comment]
Thanks for reaching out to help, really appreciate it!

So I tried both, individually

```
mosquitto_pub -h localhost -t 'zigbee2mqtt/Fstr_Vorratskeller/set' -u xxx -P xxx -d -m '{
  "configure_j1": {
     "windowCoveringType": 0,
     "open_to_closed_s": 15,
     "closed_to_open_s": 15
  }
}'
```

as well as

```
mosquitto_pub -h localhost -t 'zigbee2mqtt/Fstr_Vorratskeller/set' -u fhem -P piekspieks -d -m '{
  "configure_j1": {
     "calibrate": 1
  }
}'
```

Neither of this changes the behavior of the J1. Even if I go there and press the relais button - short or loooooong -, all the window motor does is move for a fraction of a second. Wattage measured by the J1 goes briefly into the 1W area and then back to 0. Very rarely, wattage moves up to 7W.

I can't find anything useful in the z2m log, either:

```
info  2023-02-08 07:34:27: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:27: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:32: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:32: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:37: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:37: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:42: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":7,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:42: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":7,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:47: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:47: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:52: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:52: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:34:58: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:35:59: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:35:59: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:04: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:04: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:09: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:09: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:14: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:14: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:19: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":5,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:19: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":5,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:24: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:24: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:36:30: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":0,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:50: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":0,"position":100,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:55: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":7,"position":94,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:55: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":10,"position":94,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:55: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":7,"position":94,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:55: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":10,"position":94,"power":1,"state":"OPEN","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:38:56: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":10,"position":0,"power":1,"state":"CLOSE","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:39:00: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":3,"position":0,"power":0,"state":"CLOSE","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
info  2023-02-08 07:39:00: MQTT publish: topic 'zigbee2mqtt/Fstr_Vorratskeller', payload '{"energy":0,"linkquality":18,"position":0,"power":0,"state":"CLOSE","update":{"installed_version":26280461,"latest_version":26280461,"state":"idle"}}'
```

FYI, if I swap out the J1 and put a Shelly Plus 2PM in, the window is moving as it should be (although calibration isn't working with the Shelly, either, because of the low wattage used by the motor). Unfortunately, the Shelly is too big for the available space. The J1 fits, I just need to get it to work...

Any other idea?

Again, appreciate the help!

**Comment by sjorge** [comment]
You still need to set the windowCoveringType when doing an auto calibration:

```
mosquitto_pub -h localhost -t 'zigbee2mqtt/Fstr_Vorratskeller/set' -u fhem -P piekspieks -d -m '{
  "configure_j1": {
     "windowCoveringType": 0,
     "calibrate": 1
  }
}'
```

Bit I don't think that's going to be the issue, you said it was low voltage so it might be tripping the stall sensor thingy on the J1.

If you're currently powering over DC you are probably hitting this https://www.ubisys.de/wp-content/uploads/ubisys-j1-technical-reference.pdf section 4.2 

If it is AC powered, you might try changing (page 17, same document)
<img width="864" alt="Screenshot 2023-02-08 at 08 46 08" src="https://user-images.githubusercontent.com/379665/217466456-2a033ff5-35d3-41d6-9641-baf9de1f99d3.png">

I'm not 100% sure z2m supports that attribute though. You should still be able to write/read from it using the dev console.

**Reply by Sailing74** [reply]
Thanks, I guess I will need to write 190 to that register 0x1006.

Never used the dev console, though. So will first need to figure out how to do that. Will report back.

**Comment by Sailing74** [comment]
Ok, it looks like I'd be trying to write to endpoint 1, cluster 0x0102 (Windows Covering - as per the cluster map in z2m called "closuresWindowCovering"). Attribute is 0x1006, and I should be writing 0x0190 to it.

In the dev console, I cannot specify hex values, only strings. The right cluster appears to be "Closures", and it knows an attribute called "ubisysInactivePowerThreshold". But when I try to write to that, I'm only getting an error ("UNSUPPORTED_ATTRIBUTE"):

`error 2023-02-08 14:03:08Publish 'set' 'write' to 'Fstr_Vorratskeller' failed: 'Error: Write 0x001fee00000064b1/1 closuresWindowCovering({"ubisysInactivePowerThreshold":190}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'`

So I tried to compile this into a command, where I can specify hex values:
```
endpoint: 1
cluster: 0x0102
command: 0x02 (this is "write", right?)
payload: {"attrId": 0x1006, "dataType": 0x21, "attrData": 0x0190}
```

(I think 0x21 is dataType unsigned int 16...)

However, the dev console won't let me execute this. Apparently the format is wrong.

If there is a description of the proper format, I'd appreciate a hint.

**Reply by sjorge** [reply]
It's probably best to use the write converter directly (what the frontend dev tab is using)

https://github.com/Koenkk/zigbee-herdsman-converters/blob/4e25863e0982cc0f7e6cb41468d412d97663da4c/converters/toZigbee.js#L39-L49

You probably also need to set the manufacturerCode as it's a ubisys specific attribute https://github.com/Koenkk/zigbee-herdsman/blob/a1c2ac429ef75a174f804d132aa41b17c2fb7fc6/src/zcl/definition/manufacturerCode.ts#L475

I didn't test this but something like

`mosquitto_pub -t zigbee2mqtt/DEV/set/write -m '{"cluster": "closuresWindowCovering", "payload": {"ubisysInactivePowerThreshold": 400}, "options": {"manufacturerCode": 4338}}'`

IIRC you need to covert the hex to decimal with write

**Comment by Sailing74** [comment]
Thanks for your help, @sjorge, which has really helped a lot to get the J1 working with z2m.

The only thing missing now is to change the button behavior: I'd like the blind to move fully up/down on a short press, and move while pressed for long presses. I believe this is covered in Section 7.5.5.2. I do have the bytes that need to be written, but it would be great if you'd have a hint on how I could get those bytes written:

<img width="790" alt="grafik" src="https://user-images.githubusercontent.com/5119859/218806884-86875519-25e3-4652-8148-26fb76fce650.png">
<img width="793" alt="grafik" src="https://user-images.githubusercontent.com/5119859/218806082-972459fa-4d96-4d70-b471-0c5b01d05e79.png">

So I'd need to write to this 232 application, cluster "Device Setup" (0xfc00) these bytes: 0x41 0x06 0x00 0x06 0x00 0x07 0x02 0x02 0x01 0x00 0x06 0x00 0x06 0x02 0x02 0x01 0x00 0x06 0x00 0x0B 0x02 0x02 0x01 0x02 0x06 0x01 0x07 0x02 0x02 0x01 0x01 0x06 0x01 0x06 0x02 0x02 0x01 0x01 0x06 0x01 0x0B 0x02 0x02 0x01 0x02

Maybe this will need to be preceeded with 0x48 to specify that an array of raw data is being sent.

I hope my question makes sense, and even more that you might be able to help with this!

**Reply by sjorge** [reply]
I can give you the config I'm using, you can probably adapt from there.

This one does up or down the full lenght on a short press.
```
export ZIGBEE_DEV=cover/office/window_front
mosquitto_pub -t zigbee2mqtt/${ZIGBEE_DEV}/set/configure_device_setup -m '{"input_actions": [[0,13,2,2,1,0],[1,13,2,2,1,1]]}'
```
Full up/down on a short press, Move Up/down on long press until released.
```
export ZIGBEE_DEV=cover/office/window_front
mosquitto_pub -t zigbee2mqtt/${ZIGBEE_DEV}/set/configure_device_setup -m '{"input_actions": [[0,13,2,2,1,0],[0,11,2,2,1,2],[1,13,2,2,1,1],[1,11,2,2,1,2]]}'
```

I think the 2nd one is what you want.

**Comment by Sailing74** [comment]
Terrific, thanks!
Any way I could buy you a beer in return?
